### DISCUSSION TITLE
Feature Request: Cancel-Waiting

### DISCUSSION THREAD

**Discussion by aovens-quantifi** [community/community#37599] [Actions]
When specifying concurrency in Github actions, I'd like to be able to specify `cancelWaiting: true`. Basically what this would do would cancel any workflows/jobs/steps for that concurrency group that are currently waiting, so that only one execution (the latest one) would be in the waiting state at any given time. AWS CodePipeline does this at the "stage" level where only the latest successful execution of the previous stage will go on to the next stage.

An example:

We have a workflow that tests and deploys the latest main to a staging environment. Assume 5 PR's are merged back to back. With `cancelWaiting`, the first merged PR would run through the workflow, the 2nd, 3rd and 4th PR's would be cancelled, and the 5th PR (containing 2, 3, 4) would run through once the first completes. This avoids a huge backlog waiting to get deployed and yet ensures deployments are happening (`cancelInProgress` could lead to a situation where you don't deploy all day if PR's get merged in at a regular cadence)

**Comment by ossanna16** [comment]
Hi there @aovens-quantifi and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Reply by aovens-quantifi** [reply]
Now that I see it running, it appears the regular "concurrency" works this way?
